src/display/editor/drawers/freedraw.js

```diff
@@ -681,14 +328,53 @@ class FreeOutliner {
       outline[i + 1] = top[i + 1];
     }
 
+    N = this._getOutlineEnd(outline, N);
+
+    for (let i = bottom.length - 6; i >= 6; i -= 6) {
+      for (let j = 0; j < 6; j += 2) {
+        if (isNaN(bottom[i + j])) {
+          outline[N] = outline[N + 1] = NaN;
+          N += 2;
+          continue;
+        }
+        outline[N] = bottom[i + j];
+        outline[N + 1] = bottom[i + j + 1];
+        N += 2;
+      }
+    }
+
+    this._getOutlineStart(outline, N);
+
+    return this.newFreeDrawOutline(
+      outline,
+      points,
+      this.#box,
+      this.#scaleFactor,
+      this.#innerMargin,
+      this.#isLTR
+    );
+  }
+
+  _getOutlineTwoPoints(points) {
```

Comment:
Can we make this method actually private instead?
```suggestion
  #getOutlineTwoPoints(points) {
```